tunnel-server/src/seesion.ts

```diff
@@ -0,0 +1,35 @@
+import { IncomingMessage, ServerResponse } from 'http'
+import Cookies from 'cookies'
+import { randomBytes } from 'crypto'
+import * as z from "zod"
+
+// for testing, for production workload use the env var COOKIE_SECRETS
+function generateSecret(){
+  return randomBytes(32)
+    .toString('base64')
+    .slice(0, 32)
+}
+
+export function sessionManager<T>(opts: {domain: string, schema: z.ZodSchema<T>, keys?: string[] }){
```

Comment:
nit: `Xmanager` is a naming anti-pattern, how about just `session`?

Reply:
It's to represent a mechanism for creating a factory for extracting a stateful session from a request, so simply 'session' is less suited here.
Manager makes sense IMO, but I'm open to a less generic name.